torchaudio/datasets/librispeech.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+import torch.utils.data as data
+
+import torchaudio
+from torchaudio.datasets.utils import download, extract, shuffle, walk
+
+
+def load_librispeech_item(fileid, path, ext_audio, ext_txt):
+
+    speaker, chapter, utterance = fileid.split("-")
+
+    file_text = speaker + "-" + chapter + ext_txt
+    file_text = os.path.join(path, speaker, chapter, file_text)
+    file_audio = speaker + "-" + chapter + "-" + utterance + ext_audio
+    file_audio = os.path.join(path, speaker, chapter, file_audio)
+
+    # Load audio
+    waveform, sample_rate = torchaudio.load(file_audio)
+
+    # Load text
+    for line in open(file_text):
+        fileid_text, content = line.strip().split(" ", 1)
+        if file_audio == fileid_text:
+            break
+    else:
+        # Translation not found
+        raise ValueError
+
+    return {
+        "speaker": speaker,
+        "chapter": chapter,
+        "utterance": utterance,
+        "content": content,
+        "waveform": waveform,
+        "sample_rate": sample_rate,
+    }
+
+
+class LIBRISPEECH(data.IterableDataset):
+
+    _ext_txt = ".trans.txt"
+    _ext_audio = ".flac"
+
+    def __init__(self, root, selection):
+
+        selections = [
+            "dev-clean",
+            "test-clean",
+            "test-other",
+            "train-clean-100",
+            "train-clean-360",
+            "train-other-500",
+        ]
+
+        assert selection in selections
+
+        ext_archive = ".tar.gz"
+        base = "http://www.openslr.org/resources/12/"
```

Comment:
I suggest we split this out into a URL constant within the file that's referenced as the default value in a kwarg, but the user should be able to specify e.g. a mirror on S3 or within an internal host or maybe even a file